internal/pkg/service/stream/definition/repository/sink/sink_exists.go

```diff
@@ -0,0 +1,29 @@
+package sink
+
+import (
+	serviceError "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+)
+
+func (r *Repository) ExistsOrErr(k key.SinkKey) *op.TxnOp[op.NoResult] {
+	return op.Txn(r.client).
+		Merge(r.sources.ExistsOrErr(k.SourceKey)).
+		Merge(r.schema.Active().ByKey(k).Exists(r.client).
+			WithEmptyResultAsError(func() error {
+				return serviceError.NewResourceNotFoundError("sink", k.SinkID.String(), "source")
+			}),
+		).
+		FirstErrorOnly()
+}
+
+func (r *Repository) MustNotExists(k key.SinkKey) *op.TxnOp[op.NoResult] {
```

Comment:
exported but used internally. Not covered by tests as standalone usage. Is it intended?